settings/settings.ts

```diff
@@ -0,0 +1,30 @@
+export interface BlockrefferSettings {
+	format: string;                 // Format string for link insert.
+	keepText: boolean;              // Keep selected text as link display text
+	parseLinks: boolean;            // Should we parse links with unlinkfy?
+	selectedTextAsSearch: boolean;  // Use selected text as inital search value
+    toSearch: {
+        content: boolean, 
+        path: boolean, 
+        id: boolean
+    };                              // What should we search for inside files
+    searchLimit: number;            // The limit of search results to display
+    removeIdFromContent: boolean;   // Should we remove the ^block-id
+    fileName: string;               // Should we display the file as it's path or it's name. 
+                                    //   Can be "path" or "base"
+}
```

Comment:
opportunity for some tsdocs

```suggestion
export interface BlockrefferSettings {
    /** The format string for inserting a backlink.
     * @example `!{link}` becomes `![[link]]
     */
    format: string;

    /** Whether the selected text should be retained as the link display text. */
    keepText: boolean;

    /** Whether links should be parsed using the `unlinkfy` method. */
    parseLinks: boolean;

    /** Uses the selected text as the initial search value. */
    selectedTextAsSearch: boolean;

    /** Specifies content that should be searched. */
    toSearch: {
        /** Whether to search within file content. */
        content: boolean;

        /** Whether to search for matches in the file path. */
        path: boolean;

        /** Whether to search for matches by file ID. */
        id: boolean;
    };

    /** Maximum number of search results to display. */
    searchLimit: number;

    /** Whether to remove `^block-id` from the content. */
    removeIdFromContent: boolean;

    /** How the file name should be displayed: "path" for full path or "base" for name only. */
    fileName: "path" | "base";
}
```